### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by what-the-hub**
### create a decorator

**Create `validate` method decorator**
**@param:** regex

The decorator validates method param. If it's valid, it calls the initial method. If it's not, throw console error that arguments are not valid.
Created a class with methods: 
`sendEmail (email: string)`, 
`call (phone: string)`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,77 +9,43 @@ var __metadata = (this && this.__metadata) || function (k, v) {
     if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
 };
 console.log('hello');
```
Comment (Kregopaulgue) [inline index.js]: ```Index.js``` has to be added to ```.gitignore```, as it's a compiled file

**Review by Kregopaulgue — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -9,77 +9,43 @@ var __metadata = (this && this.__metadata) || function (k, v) {
...
-    constructor(public name: string, public age: number) {
-        console.log('Some text')
-    }
-}*/
-function addShow(target, propertyKey) {
-    throw new Error('something wrong');
-    //console.log('something')
+const emailReg = /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+const phoneRegex = /^\+?3?8?(0\d{9})$/;
+const checkRegex = (email, regExp) => {
```
Comment (silchencko) [inline index.js]: not email only, but phone number too. Could you rename it?

**Review by silchencko — COMMENTED** [review]
Thanks for a good job, @what-the-hub @Kregopaulgue!

File: index.js (hunk 1 above)
Comment (what-the-hub) [reply index.js]: @Kregopaulgue thanks! I`ll fix it

**Comment by what-the-hub** [comment]
@silchencko @Kregopaulgue refactored code according yours comments. Thanks

**Review by silchencko — APPROVED** [review]
Thanks for README!

**Pull request merged**